src-java/flowhs-topology/flowhs-storm-topology/src/test/java/org/openkilda/wfm/topology/flowhs/validation/FlowValidatorTest.java

```diff
@@ -258,46 +269,98 @@ public void checkForEncapsulationTypeRequirementCorrectTypeTest() throws Invalid
         flowValidator.checkForEncapsulationTypeRequirement(SRC_ENDPOINT, properties, TRANSIT_VLAN);
     }
 
-    private RequestedFlow getTestRequestWithMaxLatencyAndMaxLatencyTier2(Long maxLatency, Long maxLatencyTier2) {
-        return RequestedFlow.builder()
+    @Test(expected = InvalidFlowException.class)
+    public void failIfMaxLatencyTier2HigherThanMaxLatencyTest() throws InvalidFlowException {
+        RequestedFlow flow = getTestRequestWithMaxLatencyAndMaxLatencyTier2((long) 1000, (long) 500);
```

Comment:
Some people consider casting as an anti-pattern. For literals you can use L: `1000L`, for other cases there is parsing methods or valueOf in the wrapper classes

Reply:
The old code will update. Thanks!

Reply:
Fixed.